source/FreeRTOS_Routing.c

```diff
@@ -761,11 +760,7 @@ void FreeRTOS_FillEndPoint( NetworkInterface_t * pxNetworkInterface,
 
                 if( xSameMACAddress == pdTRUE )
                 {
-                    if( ( pxFound[ rMATCH_MAC_ADDR ] == NULL ) || ( xIsIPv6 != ( uint32_t ) pxFound[ rMATCH_MAC_ADDR ]->bits.bIPv6 ) )
```

Comment:
Hey Hein, is there any reason to remove this condition?

Reply:
That is because I found later that the endpoint selection should never return an endpoint of the wrong IP-type.
~~~c
if( xIsIPv6 == ( BaseType_t ) pxEndPoint->bits.bIPv6 )
{
    ...
    if( xSameMACAddress == pdTRUE )
    {
        xCount[ rMATCH_MAC_ADDR ]++;
        pxFound[ rMATCH_MAC_ADDR ] = pxEndPoint;
    }
}
~~~
So from now on, a matching MAC address is only valid if the endpoint has the right IP-type.